src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -132,6 +131,11 @@ export class UiSettingsClient implements IUiSettingsClient {
     return this.overrides.hasOwnProperty(key);
   }
 
+  isSensitive(key: string): boolean {
+    const definition = this.defaults[key];
+    return !!definition?.sensitive;
```

Comment:
NIT (and personal opinion): Now that we have the `??` operator, I think `definition?.sensitive ?? false` would be slightly more explicit.

Reply:
I might change my opinion about this in the future but the I think using `??` with `false` is redundan. I find `!!` more explicit and succent